code/plotting.py

```diff
@@ -377,15 +377,41 @@ def do_plot(
+    best_fit_expected_kin = 'best_fit_expected_'+plotting_method+'_kin'
+
+    if(('sally' in methods) and ('sallino' not in methods)):
+        plot_input = np.array([
+            [variables_to_plot['p_values_expected_rate'], variables_to_plot['best_fit_expected_rate'], "Rate", "black", "solid"],
+            [variables_to_plot['p_values_expected_histo'], variables_to_plot['best_fit_expected_histo'], "Histo", "limegreen", "solid"],
+            [variables_to_plot['p_values_expected_histo_kin'], variables_to_plot['best_fit_expected_histo_kin'], "Histo-Kin", "limegreen", "dashed"],
+            [variables_to_plot['p_values_expected_sally'], variables_to_plot['best_fit_expected_sally'], "SALLY", "C0", "solid"],
+            [variables_to_plot['p_values_expected_sally_kin'], variables_to_plot['best_fit_expected_sally_kin'], "SALLY-Kin", "C0", "dashed"],
+            [variables_to_plot[p_values_expected], variables_to_plot[best_fit_expected], plotting_method.upper(), "red", "solid"],
+            [variables_to_plot[p_values_expected_kin], variables_to_plot[best_fit_expected_kin], plotting_method.upper()+"-Kin", "red","dashed"],
+        ])
+
+    elif(('sally' not in methods) and ('sallino' in methods)):
+        plot_input = np.array([
+            [variables_to_plot['p_values_expected_rate'], variables_to_plot['best_fit_expected_rate'], "Rate", "black", "solid"],
+            [variables_to_plot['p_values_expected_histo'], variables_to_plot['best_fit_expected_histo'], "Histo", "limegreen", "solid"],
+            [variables_to_plot['p_values_expected_histo_kin'], variables_to_plot['best_fit_expected_histo_kin'], "Histo-Kin", "limegreen", "dashed"],
+            [variables_to_plot['p_values_expected_sallino'], variables_to_plot['best_fit_expected_sallino'], "SALLINO", "C0", "solid"],
+            [variables_to_plot['p_values_expected_sallino_kin'], variables_to_plot['best_fit_expected_sallino_kin'], "SALLINO-Kin", "C0", "dashed"],
+            [variables_to_plot[p_values_expected], variables_to_plot[best_fit_expected], plotting_method.upper(), "red", "solid"],
+            [variables_to_plot[p_values_expected_kin], variables_to_plot[best_fit_expected_kin], plotting_method.upper()+"-Kin", "red","dashed"],
+        ])
+
+    else:
+        plot_input = np.array([
+            [variables_to_plot['p_values_expected_rate'], variables_to_plot['best_fit_expected_rate'], "Rate", "black", "solid"],
+            [variables_to_plot['p_values_expected_histo'], variables_to_plot['best_fit_expected_histo'], "Histo", "limegreen", "solid"],
+            [variables_to_plot['p_values_expected_histo_kin'], variables_to_plot['best_fit_expected_histo_kin'], "Histo-Kin", "limegreen", "dashed"],
+            [variables_to_plot[p_values_expected], variables_to_plot[best_fit_expected], plotting_method.upper(), "red", "solid"],
+            [variables_to_plot[p_values_expected_kin], variables_to_plot[best_fit_expected_kin], plotting_method.upper()+"-Kin", "red","dashed"],
```

Comment:
These 5 exact lines are added to the `plot_input` array in **every** if-else clause.

It would be better to remove this `else` clause as well as the inclusion of these lines on the other `if` / `elif` clauses, so that they can be appended before and after the conditional block:

```python3
plot_input_list = [
    [variables_to_plot['p_values_expected_rate'], variables_to_plot['best_fit_expected_rate'], "Rate", "black", "solid"],
    [variables_to_plot['p_values_expected_histo'], variables_to_plot['best_fit_expected_histo'], "Histo", "limegreen", "solid"],
    [variables_to_plot['p_values_expected_histo_kin'], variables_to_plot['best_fit_expected_histo_kin'], "Histo-Kin", "limegreen", "dashed"],
]

if (...) and (...):
    ...

elif (...) and (...):
    ...

plot_input = np.array(
    *plot_input_list,
    [variables_to_plot[p_values_expected], variables_to_plot[best_fit_expected], plotting_method.upper(), "red", "solid"],
    [variables_to_plot[p_values_expected_kin], variables_to_plot[best_fit_expected_kin], plotting_method.upper()+"-Kin", "red","dashed"],
])
```